src/AudioIO.cpp

```diff
@@ -380,6 +404,27 @@ AudioIO::~AudioIO()
    mThread.reset();
 }
 
+RealtimeEffectState *AudioIO::AddState(Track *pTrack, const PluginID & id)
+{
+   if (mpState && mpState->pRealtimeScope)
```

Comment:
My usual test: two mono tracks and an EffBassTreble, then hit play, place breakpoint here.
When it breaks here, mpState is reported as empty, so no addition of EffectStates will be made. This will result in the tracks not being processed by the effect.

Reply:
I think I found the reason for this bug: mpState is not initialized until AudioIO::StartStream is reached, but that happens in response to pressing the play button, while the code above using mpState is triggered when instantiating an effect.

Reply:
AddState needs to suspend conditionally, only in case playback is happening.  That was my mistake.